src/OSX/Avalonia.MonoMac/TopLevelImpl.cs

```diff
@@ -299,10 +304,17 @@ public override void ScrollWheel(NSEvent theEvent)
 
             public override void MouseExited(NSEvent theEvent)
             {
+                _isMouseOver = false;
                 MouseEvent(theEvent, RawMouseEventType.LeaveWindow);
                 base.MouseExited(theEvent);
             }
 
+            public override void MouseEntered(NSEvent theEvent)
+            {
+                _isMouseOver = true;
```

Comment:
We might want to call _cursor.Set here

Reply:
Or even call UpdateCursor

Reply:
That should not be necessary. If I understand the documentation of NSView correctly, `AddCursorRect` should update the cursor once the mouse enters the given frame.
The problem was if the cursor was already within the frame, the cursor was not updated. Thats why i added the `_cursor.Set()`